vllm/v1/engine/output_processor.py

```diff
@@ -328,7 +328,7 @@ def process_outputs(
             # 2) Detokenize the token ids into text and perform stop checks.
             stop_string = req_state.detokenizer.update(
                 new_token_ids, finish_reason == FinishReason.STOP)
-            if stop_string and finish_reason != FinishReason.STOP:
+            if stop_string:
```

Comment:
A dumb question: how can we skip this check?

Reply:
This is already accounted for inside the `update` method. It won't even check for stop strings in the case that `finish_reason == FinishReason.STOP` (i.e. if the engine has already stopped due to eos or a stop token).

Reply:
Got it. Thanks!